go/vt/vtgate/vcursor_impl.go

```diff
@@ -1072,6 +1088,13 @@ func (vc *vcursorImpl) keyForPlan(ctx context.Context, query string, buf io.Stri
 	_, _ = buf.WriteString(vindexes.TabletTypeSuffix[vc.tabletType])
 	_, _ = buf.WriteString("+Collate:")
 	_, _ = buf.WriteString(collations.Local().LookupName(vc.collation))
+	// The plans are going to be different based on the foreign key checks state. So we need to use that value
+	// as part of the plan's hashing key.
+	fkChecksState := vc.GetForeignKeyChecksState()
+	if fkChecksState != nil {
+		_, _ = buf.WriteString("+fkChecksState:")
+		_, _ = buf.WriteString(sqlparser.FkChecksStateString(fkChecksState))
```

Comment:
foreign_key checks become part of the query as a query comment hint (during rewriter phase), does it need to be part of the plan prefix key again?

Reply:
Yeah, you're right. Fixed it now.

Reply:
Good catch I hadn't thought of this!